presto-hive/src/main/java/com/facebook/presto/hive/util/SerDeUtils.java

```diff
@@ -216,12 +216,13 @@ private static Block serializeMap(Type type, BlockBuilder builder, Object object
         ObjectInspector keyInspector = inspector.getMapKeyObjectInspector();
         ObjectInspector valueInspector = inspector.getMapValueObjectInspector();
         BlockBuilder currentBuilder;
-        if (builder != null) {
-            currentBuilder = builder.beginBlockEntry();
-        }
-        else {
-            currentBuilder = new InterleavedBlockBuilder(typeParameters, new BlockBuilderStatus(), map.size());
+
+        boolean builderSynthesized = false;
```

Comment:
I understand that this is a general comment and not something this pull request needs to change. Let me know if i'm wrong.